packaging/macos/lib/installer_lib.sh

```
@@ -0,0 +1,311 @@
+        /opt/homebrew/lib/node_modules/@openai/codex/package.json; do
+        [[ -f "${pkg}" ]] || continue
+        local v; v="$(_read_json_version "${pkg}")"
+        if [[ -n "${v}" ]]; then echo "${v}"; return; fi
+      done
+      # Homebrew cask keeps the binary under Caskroom with a version
+      # in the path itself: /opt/homebrew/Caskroom/codex/<version>/...
+      local caskroom
+      for caskroom in /opt/homebrew/Caskroom/codex /usr/local/Caskroom/codex; do
+        if [[ -d "${caskroom}" ]]; then
+          local ver
+          ver="$(ls -1 "${caskroom}" 2>/dev/null | grep -E '^[0-9]+\.[0-9]+' | sort -V | tail -1)"
```

Shellcheck (0.11.0)

[warning] Don't use ls | grep. Use a glob or a for loop with a condition to allow non-alphanumeric filenames.

(SC2010)